### PR TITLE
add  keras2  GlobalAveragePooling layer and test

### PR DESCRIPTION
**Pull request by yjjiang11**
This  layer has passed all the tests .The test number on JKs is 663. @zhichao-li

### CODE REVIEW

**Comment by zhichao-li** [comment]
cc @simanw304

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling1D.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  *
+  * When you use this layer as the first layer of a model, you need to provide the argument
+  * inputShape (a Single Shape, does not include the batch dimension).
+  *
+  * @param inputShape A Single Shape, does not include the batch dimension.
+  * @tparam T The numeric type of parameter(e.g. weight, bias). Only support float/double now.
+  */
+class GlobalAveragePooling1D[T: ClassTag](
+                                           override val inputShape: Shape = null)(implicit ev: TensorNumeric[T])
+  extends GlobalPooling1D[T](inputShape) with Net {
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/GlobalAveragePooling1D.scala]: can we inherit from `klayers1.GlobalAveragePooling1D` directly?

**Comment by zhichao-li** [comment]
add python API?

**Comment by hkvision** [comment]
jenkins passed http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Validation/663/

File: pyzoo/zoo/pipeline/api/keras2/layers/pooling.py [hunk 2]
```diff
@@ -57,3 +57,43 @@ def __init__(self,
...
                                            list(input_shape) if input_shape else None,
                                            **kwargs)
+
+
+class GlobalAveragePooling1D(ZooKeras2Layer):
+
+    """Global average pooling operation for spatial data.
+
+    # Arguments
+        data_format: A string,
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras2/layers/pooling.py]: wrong doc?

File: pyzoo/zoo/pipeline/api/keras2/layers/pooling.py [hunk 3]
```diff
@@ -57,3 +57,43 @@ def __init__(self,
...
+            one of `channels_last` (default) or `channels_first`.
+            The ordering of the dimensions in the inputs.
+            `channels_last` corresponds to inputs with shape
+            `(batch, height, width, channels)` while `channels_first`
+            corresponds to inputs with shape
+            `(batch, channels, height, width)`.
+            It defaults to the `image_data_format` value found in your
+            Keras config file at `~/.keras/keras.json`.
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras2/layers/pooling.py]: do not copy from keras doc... copy from analytics zoo keras1 doc...

**Comment by hkvision** [comment]
Need to rebase and resolve conflicts :)
Then run jenkins again

**Pull request closed without merging**